src/lib/storage/chunk.cpp

```diff
@@ -1,31 +1,56 @@
 #include "chunk.hpp"
+#include <boost/hana/for_each.hpp>
 
 #include "abstract_segment.hpp"
 #include "utils/assert.hpp"
+#include "value_segment.hpp"
 
 namespace opossum {
 
 void Chunk::add_segment(const std::shared_ptr<AbstractSegment> segment) {
-  // Implementation goes here
+  DebugAssert(_segments.size() < std::numeric_limits<ColumnCount>::max(), "Segment limit is already reached.");
+  _segments.emplace_back(segment);
 }
 
 void Chunk::append(const std::vector<AllTypeVariant>& values) {
-  // Implementation goes here
+  auto size = _segments.size();
+  DebugAssert(values.size() == size, "Number of values and number of columns should be equal.");
+
+  for (size_t i = 0; i < size; ++i) {
```

Comment:
Firstly, you could think about adapting the code style described in the lecture today, using `auto x = type{y}` and not using `i` as count variable. 

Secondly, we could reduce the LOC by one, by removing the `auto size = _segments.size();` statement. Luckily, we do not sacrifice computational power, since `vector::size` has a [constant complexity](https://en.cppreference.com/w/cpp/container/vector/size).

```suggestion
  DebugAssert(values.size() == _segments.size(), "Number of values and number of columns should be equal.");

  for (auto seg_index = uint32_t{0}; seg_ind < segments.size(); ++seg_ind) {
```